pipeline_steps/common.groovy

```diff
@@ -1511,13 +1511,61 @@ def findHookDir(String hook_dirs) {
 }
 
 void runReleasesPullRequestWorkflow(String baseBranch, String prBranch, String jiraProjectKey){
-  (prType, componentText) = getComponentChange(baseBranch, prBranch)
+  def (prType, componentText) = getComponentChange(baseBranch, prBranch)
+  def componentYaml = readYaml text: componentText
 
   if (prType == "release"){
-    testRelease(component_text)
-    createRelease()
+    String REPO_NAME=componentYaml["name"]
+    String REPO_URL=componentYaml["repo_url"]
+    String SHA=componentYaml["release"]["sha"]
+    String MAINLINE=componentYaml["release"]["series"]
+    String RC_BRANCH="${MAINLINE}-rc"
+
+    dir(REPO_NAME){
+      clone_with_pr_refs(
+        "./",
+        REPO_URL,
+        "master",
+      )
+      println "=== Checking that the specified SHA exists ==="
+      Integer no_sha = sh(
+        script: """#!/bin/bash -xe
+          git rev-parse --verify ${SHA}^{commit}
+        """,
+        returnStatus: true,
+      )
+      if (no_sha) {
+        throw new Exception("The supplied SHA ${SHA} was not found.")
+      }
+      println "=== Checking for the existence of an RC branch ==="
+      Integer no_rc = sh(
```

Comment:
This is only used negatively eg `!no_rc` which is effectively `not no rc`, so for simplicity maybe it should just be `rc`?